CMake fails if the latest tag is a non-version git tag

**Describe the bug**
```
CMake Error at cmake/version.cmake:29 (message):
  Git repository does not appear to contain any tag starting with ccf- (the
  repository should be cloned with sufficient depth to access the latest
  "ccf-*" tag)
```

**To Reproduce**
```
git tag not-a-ccf-version
cmake ..
```

**Expected behavior**
Should find the latest version-like tag, not any tag.

--- a/cmake/version.cmake
+++ b/cmake/version.cmake
@@ -10,7 +10,7 @@ if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
   find_package(Git)
 
   execute_process(
-    COMMAND "bash" "-c" "${GIT_EXECUTABLE} describe --tags"
+    COMMAND "bash" "-c" "${GIT_EXECUTABLE} describe --tags --match=\"ccf-*\""
     OUTPUT_VARIABLE "CCF_VERSION"
     OUTPUT_STRIP_TRAILING_WHITESPACE
     RESULT_VARIABLE RETURN_CODE

